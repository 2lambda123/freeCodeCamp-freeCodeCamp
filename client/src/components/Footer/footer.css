/* 11. Site Footer
/* ---------------------------------------------------------- */

.site-footer {
  color: var(--tertiary-color);
  background: var(--tertiary-background);
  line-height: 1.6;
  padding: 40px 15px;
  font-size: 16px;
}

.footer-top {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(18em, 1fr));
}
.footer-top,
.footer-bottom {
  width: min(100%, 1200px);
  margin-inline: auto;
}

.site-footer p {
  margin: 0 0 1.45rem;
  line-height: 30px;
  font-size: 16px;
}

.site-footer a {
  color: var(--tertiary-color);
  text-decoration: none;
}

.site-footer a:hover {
  text-decoration: underline;
}

.site-footer .col-header {
  text-align: center;
  font-weight: 700;
  font-size: 16px;
  padding: 0 15px 15px;
  margin-bottom: 0;
}

.trending-guides-articles {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(10em, 1fr));
  gap: 0.5rem;
}

.footer-row {
  margin: 0;
}

.footer-col {
  padding-inline: 15px;
  font-size: 16px;
}

.footer-col a {
  padding: 5px 0;
}

.footer-desc-col {
  padding-inline: 15px;
  margin-bottom: 30px;
}

.footer-desc-col a {
  text-decoration: underline;
}

p.footer-donation {
  font-weight: 700;
  font-size: 18px;
}

p.footer-donation a:hover {
  text-decoration: none;
}

.trending-guides {
  margin: 0 0 3rem;
}

.footer-bottom .our-nonprofit {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.footer-bottom .our-nonprofit a {
  padding: 5px 10px;
}

@media (min-width: 500px) {
  .footer-col {
    font-size: 15;
  }
}

@media (min-width: 800px) {
  .footer-col {
    font-size: 16.5;
  }
  .footer-bottom .our-nonprofit a {
    padding: 5px;
  }
}

@media (min-width: 1200px) {
  .footer-desc-col {
    flex: 1 0 45%;
  }
  .trending-guides {
    flex: 1 0 58%;
  }
  p.footer-donation {
    margin-top: 40px;
  }
  .footer-bottom .our-nonprofit {
    padding: 0 10px;
    justify-content: space-between;
  }
  .footer-bottom .col-header {
    visibility: hidden;
  }
  .our-nonprofit {
    margin-top: 20px;
  }
  .footer-divder {
    height: 2px;
    margin: 0 15px;
    background-color: var(--quaternary-background);
  }
}
