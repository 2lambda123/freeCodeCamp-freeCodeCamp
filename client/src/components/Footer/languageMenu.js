/* eslint-disable jsx-a11y/no-onchange */
import React from 'react';
import { useTranslation } from 'react-i18next';

const { langDisplayNames, availableLangs } = require('../../../i18n/allLangs');
const { homeLocation } = require('../../../config/env');

const locales = availableLangs.client;

const LanguageMenu = () => {
  const { i18n, t } = useTranslation();
  const currentLanguage = i18n.language;

  const changeLanguage = e => {
    // see if we can get the path from gatsby or something?
    const path = window.location.pathname;

    if (e.target.value === `english`) {
      // look into using another method here <Router>?
      window.location.replace(`${homeLocation}${path}`);
    } else if (e.target.value === 'espanol') {
      window.location.replace(`${homeLocation}/espanol${path}`);
    }
  };

  return (
    <div className='language-menu'>
      <span>{t('footer.language')}</span>
      <select onChange={e => changeLanguage(e)} value={currentLanguage}>
        {locales.map((lang, i) => {
          return (
            <option key={i} value={lang}>
              {langDisplayNames[lang]}
            </option>
          );
        })}
      </select>
    </div>
  );
};

export default LanguageMenu;
