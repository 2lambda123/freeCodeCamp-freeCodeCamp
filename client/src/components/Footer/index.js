import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import Link from '../helpers/Link';
import LanguageMenu from './LanguageMenu';
import './footer.css';

const { showLocaleDropdownMenu = false } = require('../../../config/env');

const propTypes = {
  children: PropTypes.any
};

const ColHeader = ({ children, ...other }) => (
  <div className='col-header' {...other}>
    {children}
  </div>
);
ColHeader.propTypes = propTypes;

function Footer() {
  const { t } = useTranslation();

  const trendingArticles = (() => {
    const numbers = Array(30)
      .fill(0)
      .map((_, i) => i);
    const list = numbers.map(number => (
      <Link
        external={false}
        key={number}
        to={t(`trending:articles.${number}.link`)}
      >
        {t(`trending:articles.${number}.title`)}
      </Link>
    ));
    return list;
  })();

  return (
    <footer className='site-footer'>
      <div className='footer-container'>
        <div className='footer-top'>
          <div className='footer-desc-col'>
            {showLocaleDropdownMenu ? <LanguageMenu /> : null}
            <p>{t('footer.tax-exempt-status')}</p>
            <p>{t('footer.mission-statement')}</p>
            <p>{t('footer.donation-initiatives')}</p>
            <p className='footer-donation'>
              {t('footer.donate-text')}{' '}
              <Link className='inline' to='/donate'>
                {t('footer.donate-link')}
              </Link>
              .
            </p>
          </div>
          <div className='trending-guides'>
            <div className='col-header'>{t('footer.trending-guides')}</div>
            <div className='trending-guides-row'>
              <div className='footer-col footer-col-1'>
                {trendingArticles.slice(0, 10)}
              </div>
              <div className='footer-col footer-col-2'>
                {trendingArticles.slice(10, 20)}
              </div>
              <div className='footer-col footer-col-3'>
                <div className='footer-left'>
                  {trendingArticles.slice(20, 25)}
                </div>

                <div className='footer-right'>{trendingArticles.slice(25)}</div>
              </div>
            </div>
          </div>
        </div>
        <div className='footer-bottom'>
          <div className='col-header'>{t('footer.our-nonprofit')}</div>
          <div className='footer-divder' />
          <div className='our-nonprofit'>
            <Link external={false} to={t('footer.links.about-url')}>
              {t('footer.links.about')}
            </Link>
            <Link
              external={false}
              sameTab={false}
              to={'https://www.linkedin.com/school/free-code-camp/people/'}
            >
              {t('footer.links.alumni')}
            </Link>
            <Link external={false} to={'https://github.com/freeCodeCamp/'}>
              {t('footer.links.open-source')}
            </Link>
            <Link
              external={false}
              sameTab={false}
              to={t('footer.links.shop-url')}
            >
              {t('footer.links.shop')}
            </Link>
            <Link external={false} to={t('footer.links.support-url')}>
              {t('footer.links.support')}
            </Link>
            <Link external={false} to={t('footer.links.sponsors-url')}>
              {t('footer.links.sponsors')}
            </Link>
            <Link external={false} to={t('footer.links.honesty-url')}>
              {t('footer.links.honesty')}
            </Link>
            <Link external={false} to={t('footer.links.coc-url')}>
              {t('footer.links.coc')}
            </Link>
            <Link external={false} to={t('footer.links.privacy-url')}>
              {t('footer.links.privacy')}
            </Link>
            <Link external={false} to={t('footer.links.tos-url')}>
              {t('footer.links.tos')}
            </Link>
            <Link external={false} to={t('footer.links.copyright-url')}>
              {t('footer.links.copyright')}
            </Link>
          </div>
        </div>
      </div>
    </footer>
  );
}

Footer.displayName = 'Footer';
export default Footer;
