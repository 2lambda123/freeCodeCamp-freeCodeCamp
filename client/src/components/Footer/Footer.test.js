/* global expect */
import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import Footer from './';

describe('<Footer />', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
