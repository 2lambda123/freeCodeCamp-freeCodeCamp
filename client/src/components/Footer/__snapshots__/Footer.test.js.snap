// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches snapshot 1`] = `
<footer
  className="site-footer"
>
  <div
    className="footer-container"
  >
    <div
      className="footer-top"
    >
      <div
        className="footer-desc-col"
      >
        <div
          className="language-menu"
        >
          <label>
            footer.language
            <select
              onChange={[Function]}
            >
              <option
                value="english"
              >
                English
              </option>
              <option
                value="espanol"
              >
                Español
              </option>
              <option
                value="chinese"
              >
                中文
              </option>
            </select>
          </label>
        </div>
        <p>
          footer.tax-exempt-status
        </p>
        <p>
          footer.mission-statement
        </p>
        <p>
          footer.donation-initiatives
        </p>
        <p
          className="footer-donation"
        >
          footer.donate-text
           
          <a
            className="inline"
            href="/donate"
          >
            footer.donate-link
          </a>
          .
        </p>
      </div>
      <div
        className="trending-guides"
      >
        <div
          className="col-header"
        >
          footer.trending-guides
        </div>
        <div
          className="trending-guides-row"
        >
          <div
            className="footer-col footer-col-1"
          >
            <a
              href="trending:articles.0.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.0.title
            </a>
            <a
              href="trending:articles.1.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.1.title
            </a>
            <a
              href="trending:articles.2.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.2.title
            </a>
            <a
              href="trending:articles.3.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.3.title
            </a>
            <a
              href="trending:articles.4.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.4.title
            </a>
            <a
              href="trending:articles.5.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.5.title
            </a>
            <a
              href="trending:articles.6.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.6.title
            </a>
            <a
              href="trending:articles.7.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.7.title
            </a>
            <a
              href="trending:articles.8.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.8.title
            </a>
            <a
              href="trending:articles.9.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.9.title
            </a>
          </div>
          <div
            className="footer-col footer-col-2"
          >
            <a
              href="trending:articles.10.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.10.title
            </a>
            <a
              href="trending:articles.11.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.11.title
            </a>
            <a
              href="trending:articles.12.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.12.title
            </a>
            <a
              href="trending:articles.13.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.13.title
            </a>
            <a
              href="trending:articles.14.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.14.title
            </a>
            <a
              href="trending:articles.15.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.15.title
            </a>
            <a
              href="trending:articles.16.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.16.title
            </a>
            <a
              href="trending:articles.17.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.17.title
            </a>
            <a
              href="trending:articles.18.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.18.title
            </a>
            <a
              href="trending:articles.19.link"
              rel="noopener noreferrer"
              target="_blank"
            >
              trending:articles.19.title
            </a>
          </div>
          <div
            className="footer-col footer-col-3"
          >
            <div
              className="footer-left"
            >
              <a
                href="trending:articles.20.link"
                rel="noopener noreferrer"
                target="_blank"
              >
                trending:articles.20.title
              </a>
              <a
                href="trending:articles.21.link"
                rel="noopener noreferrer"
                target="_blank"
              >
                trending:articles.21.title
              </a>
              <a
                href="trending:articles.22.link"
                rel="noopener noreferrer"
                target="_blank"
              >
                trending:articles.22.title
              </a>
              <a
                href="trending:articles.23.link"
                rel="noopener noreferrer"
                target="_blank"
              >
                trending:articles.23.title
              </a>
              <a
                href="trending:articles.24.link"
                rel="noopener noreferrer"
                target="_blank"
              >
                trending:articles.24.title
              </a>
            </div>
            <div
              className="footer-right"
            >
              <a
                href="trending:articles.25.link"
                rel="noopener noreferrer"
                target="_blank"
              >
                trending:articles.25.title
              </a>
              <a
                href="trending:articles.26.link"
                rel="noopener noreferrer"
                target="_blank"
              >
                trending:articles.26.title
              </a>
              <a
                href="trending:articles.27.link"
                rel="noopener noreferrer"
                target="_blank"
              >
                trending:articles.27.title
              </a>
              <a
                href="trending:articles.28.link"
                rel="noopener noreferrer"
                target="_blank"
              >
                trending:articles.28.title
              </a>
              <a
                href="trending:articles.29.link"
                rel="noopener noreferrer"
                target="_blank"
              >
                trending:articles.29.title
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="footer-bottom"
    >
      <div
        className="col-header"
      >
        footer.our-nonprofit
      </div>
      <div
        className="footer-divder"
      />
      <div
        className="our-nonprofit"
      >
        <a
          href="footer.links.about-url"
          rel="noopener noreferrer"
          target="_blank"
        >
          footer.links.about
        </a>
        <a
          href="https://www.linkedin.com/school/free-code-camp/people/"
          rel="noopener noreferrer"
          target="_blank"
        >
          footer.links.alumni
        </a>
        <a
          href="https://github.com/freeCodeCamp/"
          rel="noopener noreferrer"
          target="_blank"
        >
          footer.links.open-source
        </a>
        <a
          href="footer.links.shop-url"
          rel="noopener noreferrer"
          target="_blank"
        >
          footer.links.shop
        </a>
        <a
          href="footer.links.support-url"
          rel="noopener noreferrer"
          target="_blank"
        >
          footer.links.support
        </a>
        <a
          href="footer.links.sponsors-url"
          rel="noopener noreferrer"
          target="_blank"
        >
          footer.links.sponsors
        </a>
        <a
          href="footer.links.honesty-url"
          rel="noopener noreferrer"
          target="_blank"
        >
          footer.links.honesty
        </a>
        <a
          href="footer.links.coc-url"
          rel="noopener noreferrer"
          target="_blank"
        >
          footer.links.coc
        </a>
        <a
          href="footer.links.privacy-url"
          rel="noopener noreferrer"
          target="_blank"
        >
          footer.links.privacy
        </a>
        <a
          href="footer.links.tos-url"
          rel="noopener noreferrer"
          target="_blank"
        >
          footer.links.tos
        </a>
        <a
          href="footer.links.copyright-url"
          rel="noopener noreferrer"
          target="_blank"
        >
          footer.links.copyright
        </a>
      </div>
    </div>
  </div>
</footer>
`;
